packages/rs-drive-light-client/src/proof/verify.rs

```diff
@@ -0,0 +1,115 @@
+use dapi_grpc::platform::v0::{Proof, ResponseMetadata};
+use dpp::{bls_signatures, dashcore::hashes::hex::ToHex};
+pub use drive::drive::verify::RootHash;
+use tenderdash_abci::{
+    proto::types::{CanonicalVote, SignedMsgType, StateId},
+    signatures::{SignBytes, SignDigest},
+};
+
+use crate::Error;
+
+use super::from_proof::QuorumInfoProvider;
+
+pub(crate) fn verify_tenderdash_proof(
+    proof: &Proof,
+    mtd: &ResponseMetadata,
+    root_hash: &[u8],
+    provider: Box<dyn QuorumInfoProvider>,
+) -> Result<(), Error> {
+    let block_id_hash = proof.block_id_hash.to_vec();
+
+    let version = mtd.protocol_version as u64;
+    let height = mtd.height as u64;
+    let round = proof.round as u32;
+    let quorum_hash = TryInto::<[u8; 32]>::try_into(proof.quorum_hash.as_slice()).map_err(|e| {
+        Error::InvalidQuorum {
+            error: "invalid quorum hash size: ".to_string() + &e.to_string(),
+        }
+    })?;
+
+    // Now, lookup quorum details
+    let chain_id = mtd.chain_id.clone();
+    let quorum_type = proof.quorum_type;
+    let pubkey_bytes = provider.get_quorum_public_key(quorum_type, quorum_hash.to_vec())?;
```

Comment:
we should also pass the core height to verify that the quorum is still active at that height.